packages/frontend/src/ee/features/aiCopilot/utils/echarts.ts

```diff
@@ -113,68 +123,82 @@ const getTimeSeriesMetricEchartsConfig = (
-                  grid: {
-                      top: 100,
-                  },
-              }
-            : {}),
-        dataset: {
-            source: chartData,
-            dimensions: Object.keys(chartData[0] || {}),
-        },
-        animation: false,
-        backgroundColor: '#fff',
-        xAxis: [
-            {
-                type: 'time',
-            },
-        ],
-        yAxis: [
-            {
-                type: 'value',
+        type: ChartType.CARTESIAN,
+        config: {
+            layout: {
+                xField: metricQuery.dimensions[0],
+                yField: metricQuery.metrics,
             },
-        ],
-        series: metrics.map((metric: AnyType) => ({
-            type: 'line',
-            name: metric,
-            encode: {
-                x: config.xDimension,
-                y: metric,
+            eChartsConfig: {
+                ...(config.title ? { title: { text: config.title } } : {}),
+                legend: {
+                    show: true,
+                    type: 'plain',
+                },
+                grid: { containLabel: true },
+                xAxis: [
+                    {
+                        type: 'time',
+                    },
+                ] as Axis[],
+                yAxis: [
+                    {
+                        type: 'value',
+                    },
+                ] as Axis[],
+                series: config.yMetrics.map((metric) => {
+                    return {
+                        type: CartesianSeriesType.LINE,
+                        name: metric,
+                        encode: {
+                            xRef: { field: config.xDimension },
+                            yRef: { field: metric },
+                        },
+                        ...(config.lineType === 'area' && { areaStyle: {} }),
+                        yAxisIndex: 0,
+                    };
+                }),
             },
-            ...(config.lineType === 'area' && { areaStyle: {} }),
-        })),
+        },
     };
 };
 
 const getCsvMetricEchartsConfig = (
-    _config: AnyType,
+    _config: CsvFileVizConfigSchemaType,
     _rows: Record<string, unknown>[],
-) => {
-    return {};
+): ChartConfig => {
+    return {
+        type: ChartType.TABLE,
+    };
 };
 
-export const getChartOptionsFromAiAgentThreadMessageVizQuery = ({
+export const getChartConfigFromAiAgentVizConfig = ({
     config,
     rows,
     type,
+    metricQuery,
 }: {
-    config: AnyType;
+    metricQuery: MetricQuery;
     rows: Record<string, unknown>[];
-    type: AiChartType;
-}) => {
+} & (
+    | {
+          type: AiChartType.VERTICAL_BAR_CHART;
+          config: VerticalBarMetricVizConfigSchemaType;
+      }
+    | {
+          type: AiChartType.TIME_SERIES_CHART;
+          config: TimeSeriesMetricVizConfigSchemaType;
+      }
+    | {
+          type: AiChartType.CSV;
+          config: CsvFileVizConfigSchemaType;
+      }
+)): ChartConfig => {
```

Comment:
let's move this |...|...|... schema union to its own type and reference in packages/frontend/src/ee/features/aiCopilot/utils/getOpenInExploreUrl.ts?